src/lib/scheduler/operator_task.cpp

```diff
@@ -63,21 +63,13 @@ void link_tasks_for_subquery_pruning(const std::unordered_set<std::shared_ptr<Op
         const auto& subquery_root = subquery->get_or_create_operator_task();
         Assert(tasks.contains(subquery_root), "Unknown OperatorTask.");
 
-        // Cycles in the task graph would lead to deadlocks during execution. To make sure we do not introduce cycles,
-        // we only set the subquery task as predecessor of the GetTable task if it is not a successor of the GetTable
-        // task.
-        auto is_acyclic = true;
-        visit_tasks_upwards(task, [&](const auto& successor) {
-          if (successor == subquery_root) {
-            is_acyclic = false;
-            return TaskUpwardVisitation::DoNotVisitSuccessors;
-          }
-          return TaskUpwardVisitation::VisitSuccessors;
-        });
-
-        if (is_acyclic) {
-          subquery_root->set_as_predecessor_of(task);
-        }
+        // Cycles in the task graph would lead to deadlocks during execution. This could happen if a table can be pruned
```

Comment:
The entire comment assumes that StoredTableNodes/GetTables are deduplicated, which isn't said anywhere, right?